### PR TITLE
OCaml 4.11 support

### PR DESCRIPTION
**Pull request by kit-ty-kate**
~Relies on https://github.com/ocaml-ppx/ppx_deriving/pull/220~

### CODE REVIEW

**Comment by kit-ty-kate** [comment]
I've moved the commits from #111 for testing purpose and fixed a few issues there (cc @XVilka)

However the CI failed in ppxlib on 4.11. @rgrinberg @xclerc have you ever seen this error message by any chance?
```
#=== ERROR while compiling ppxlib.0.13.0 ======================================#
# context              2.0.7 | linux/x86_64 | ocaml-variants.4.11.0+trunk | git+file:///home/opam/opam-repository
# path                 ~/.opam/ocaml-variants.4.11.0+trunk/.opam-switch/build/ppxlib.0.13.0
# command              ~/.opam/ocaml-variants.4.11.0+trunk/bin/dune build -p ppxlib -j 71
# exit-code            1
# env-file             ~/.opam/log/ppxlib-312-251322.env
# output-file          ~/.opam/log/ppxlib-312-251322.out
### output ###
# [...]
#     ocamldep traverse/.ppxlib_traverse.objs/ppxlib_traverse.pp.ml.d (exit 2)
# (cd _build/default && /home/opam/.opam/ocaml-variants.4.11.0+trunk/bin/ocamldep.opt -modules -impl traverse/ppxlib_traverse.pp.ml) > _build/default/traverse/.ppxlib_traverse.objs/ppxlib_traverse.pp.ml.d
# >> Fatal error: OCaml and preprocessor have incompatible versions
# Fatal error: exception Misc.Fatal_error
```

**Comment by XVilka** [comment]
You have to drop also the 4.04 from the Travis, since new ppx_deriving-on-ppxlib supports only >=4.05.

File: ppx_deriving_yojson.opam [hunk 1]
```diff
@@ -16,10 +16,8 @@ depends: [
   "yojson" {>= "1.6.0" & < "2.0.0"}
   "result"
   "ppx_deriving" {>= "4.0" & < "5.0"}
```
Comment (XVilka) [inline ppx_deriving_yojson.opam:18]: I think also the new version restriction for ppx_deriving is required, since older versions wouldn't work.

**Comment by gasche** [comment]
I made minor review comments in #111, I don't know if they should be considered here or there.

**Comment by XVilka** [comment]
@kit-ty-kate let me know if I can help. I can continue work on this in the separate PR if you don't have time.

**Comment by thierry-martinez** [comment]
I opened a new PR https://github.com/ocaml-ppx/ppx_deriving_yojson/pull/121 to include @gasche's suggestions on https://github.com/ocaml-ppx/ppx_deriving_yojson/pull/111 and with minor changes for better handling location information when possible (`!Ast_helper.default_loc` was used but never updated).

**Comment by kit-ty-kate** [comment]
Closing in favour of #121

**Pull request closed without merging**